components/drafts/draft_actions/action.tsx

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef} from 'react';
+import {Tooltip} from 'react-bootstrap';
+import classNames from 'classnames';
+
+import OverlayTrigger from 'components/overlay_trigger';
+import Constants from 'utils/constants';
+
+import './action.scss';
+
+type Props = {
+    icon: string;
+    id: string;
+    name: string;
+    onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
+    tooltip: React.ReactNode;
```

Comment:
Is there any way to create the tooltip at this level? (and have simpler props, to avoid "same object reasignation")

Reply:
Could be a string "type" and handle the type here.

Reply:
hm, it seemed cleaner this way to me, I could change the variable from tooltip to tooltipText though

Reply:
I might be over optimizing with this, so feel free to dismiss.

Passing the "tooltipText" is "annoying", because you would need to translate it, without using the "FormattedText" component, since that would once again create an object type, and we won't resolve anything.

My idea is that in the previous component (where this component is instantiated) the tooltip is based on the type of the action, and (IIRC) it is not dependent on anything from that component, so we can just pass a string "type", and build the whole tooltip in this component.

Again, this may dirty things too much in comparison with the potential performance gain, so feel free to dismiss.